src/main/java/io/confluent/connect/elasticsearch/bulk/BulkProcessor.java

```diff
@@ -149,7 +167,7 @@ public void run() {
       wait(Math.max(0, lingerMs - elapsedMs));
```

Comment:
here, should we wait if the number of pending futures<> is larger than some limit?

Reply:
that should be limited else where by max pending requests IIRC

Reply:
looks like `maxInFlightRequests` is used to limit the number of pending requests. It does this by creating the threadpool with that number of threads. Further submits would just be queued up and we would accumulate the futures in the futures queue. I am not sure I want to enforce a hard limit on the size of the futures queue since this could break backwards compatibility when handling huge spikes of traffic that we could handle before. WDYT?